src/analyzer/protocol/quic/QUIC.spicy

```diff
@@ -298,11 +360,26 @@ type CRYPTOPayload = unit(from_client: bool) {
   cryptodata: bytes &size=self.length.result_;
 };
 
-type ACKPayload = unit {
+# https://datatracker.ietf.org/doc/html/rfc9000#ack-ranges
+type ACKRange = unit {
+  gap: VariableLengthInteger;
+  ack_range_length: VariableLengthInteger;
+};
+
+type ACKECNCounts = unit {
+  ect0: VariableLengthInteger;
+  ect1: VariableLengthInteger;
+  ecn_ce: VariableLengthInteger;
+};
+
+# https://datatracker.ietf.org/doc/html/rfc9000#name-ack-frames
+type ACKPayload = unit(frame_type: FrameType) {
   latest_ack: VariableLengthInteger;
   ack_delay: VariableLengthInteger;
   ack_range_count: VariableLengthInteger;
   first_ack_range: VariableLengthInteger;
+  ack_ranges: ACKRange[self.ack_range_count.result_];
+  ecn_counts: ACKECNCounts if(frame_type == FrameType::ACK2);
```

Comment:
Nit: for 7.2 and above we would probably write this as 
```suggestion
  if (frame_type == FrameType::ACK2) {
    ecn_counts: ACKECNCounts if(frame_type == FrameType::ACK2);
  };
```

This avoids too much "interesting" stuff happening at the very right-hand side.

Reply:
Leaving as is due to 7.0 backport candidate. The trailing `if` is something Ruby has, too, so I'm not all that upset about it once in a while. Also, whenever I see an actual `if` block, my mind still goes into "this isn't parsing code" mode. That might be personal and I'm sure I'll overcome this, but for simpler lines the trailing `if` seems totally fine.